Backpack/StarRating/Classes/BPKStar.m

```diff
@@ -0,0 +1,160 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKStar.h"
+#import <Backpack/Color.h>
+#import <Backpack/Common.h>
+#import <Backpack/Icon.h>
+#import <Backpack/BPKRTLSupport.h>
+
+@interface BPKStar ()
+
+@property (nonatomic) UIImageView *starView;
```

Comment:
Couldn't we use a single `BPKIconView` with a conditionally supplied image property. `BPKIconView` already caches the resulting `UIImage`s for performance and it would cut the number of views by half

Reply:
Sure thing, didn't know that. Also shall I implement a size setter for `BPKIconView`? I would need to be able to set it when the `BPKStar`'s size setter is called, but not sure it was intentional to keep that private.

Reply:
Also `BPKIconView` doesn't support RTL, so I'll need to add that too.

Reply:
> Also BPKIconView doesn't support RTL, so I'll need to add that too.
Why not? It's just a UIImageView it will respect UIKit's RTL controls just fine

Reply:
Yeah expose the setter, that's fine and is required for this change

Reply:
Yes, that's right. In fact it just hasn't had the ability to handle icons with a horizontal direction (flipping when necessary - e.g. a half star). I added that functionality too.